src/transfer_zip.py

```diff
@@ -0,0 +1,657 @@
+            " This is helpful for transferring a limited amount of data"
+            " during the initial testing of the deployment."
+        ),
+    )
+    parser.add_argument(
+        "--now",
+        default=None,
+        type=str,
+        help=(
+            "Now time in (ISOT, TAI timescale)."
+            " If left blank it will use astropy.time.Time.now."
+        ),
+    )
+
+    parser.add_argument(
+        "-C",
+        "--config_file",
+        required=True,
+        type=str,
+        help="Path to YAML config file.",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--dest_uri_prefix",
+        type=str,
+        default="/sdf/data/rubin/lsstdata/offline/instrument/",
+        help="Destination URI prefix for raw data.",
+    )
+
+    parser.add_argument(
+        "-r",
+        "--rucio_rse",
+        type=str,
+        required=False,
+        help="Rucio RSE for raw data.",
+    )
+    parser.add_argument(
+        "-s",
+        "--scope",
+        type=str,
+        required=False,
+        help="Rucio scope for raw data.",
+    )
+
+    parser.add_argument(
+        "--log",
+        type=str,
+        required=False,
+        help="Level name (WARNING, INFO, DEBUG) or comma-sepaarated list of logger=level pairs.",
+    )
+
+    ns = parser.parse_args()
+    ns.now = Time(ns.now, format="isot", scale="tai") if ns.now else Time.now()
+    if ns.now > Time.now():
+        raise ValueError(f"--now is in the future: {ns.now}")
+    if ns.rucio_rse is not None:
+        if ns.scope is None:
+            raise ValueError("--scope required with --rucio_rse")
+
+    return ns
+
+
+def transfer_data_query(data_query: DataQuery) -> None:
+    """Transfer all files matching a data query into zip files.
+
+    Parameters
+    ----------
+    data_query: `DataQuery`
+        The query and associated embargo time.
+    """
+    global logger, config, source_butler
+
+    # End of window is now - embargo length
+    end_time = config.now - TimeDelta(data_query.embargo_hours * 3600, format="sec")
+    # If window is defined, then start is that much before the end
+    # Otherwise, start is infinitely previous
+    if config.window is not None:
+        start_time = end_time - TimeDelta(config.window, format="quantity_str")
+    else:
+        start_time = Time(0, format="jd")
+    ok_timespan = Timespan(start_time, end_time)
+
+    # Find all exposures meeting criteria
+    dim_where = "(exposure.timespan OVERLAPS _ok_timespan)"
+    dim_where += f" AND ({data_query.where})" if data_query.where else ""
+    dim_bind = {"_ok_timespan": ok_timespan}
+    logger.info("Querying exposure: %s with %s", dim_where, dim_bind)
+    try:
+        exposures = source_butler.query_dimension_records(
+            "exposure",
+            where=dim_where,
+            bind=dim_bind,
+            limit=None,
+            instrument=data_query.instrument,
+        )
+    except EmptyQueryResultError:
```

Comment:
You can use `explain=False` if you want to not have to trap 0 records yourself.

Reply:
I still want to return early if there are no records, but at least I don't have to import the error.